### ISSUE TITLE
Query UTxOs

### ISSUE THREAD

**Issue by AhmadzadehHazhir** [input-output-hk/cardano-db-sync#1003]
when I try to query the UTxOs of an address from "utxo_view" in postgres it almost takes 1 minute , is it normal? because when I query the same address for utxos from cardano-cli it returns the result in matter of seconds or even ms.

**Comment by erikd** [comment]
Yes, the query on the database is slow, that is (unfortunately) expected.

The reason the `cardano-cli` query is quick is that its querying an in-memory data structure (as opposed to an external database) **and** the in-memory data structure holds the data in a form that makes the query much less time consuming (ie in-memory data structure does not contain spent outputs while the database does contain them).

**Comment by AhmadzadehHazhir** [comment]
thanks for the answer.

**Comment by AhmadzadehHazhir** [comment]
what kind of database does cardano-cli uses?

**Comment by erikd** [comment]
`cardano-cli` does not use a database. It uses an in-memory data structure.

**Comment by SebCli78** [comment]
Hello,
jumping here and I'm now more confused than ever...
sounds like Cardano-cli moved from in-memory to hard drive for queries
see here: https://github.com/input-output-hk/cardano-node/issues/3490

so on one side I'm asked to find other way than Cardano-cli and here it says kind of reverse

**Comment by erikd** [comment]
@SebCli78 I knew there were plans to move parts of the ledger state (the largest part of which is the UTxO set) to disk. I was not aware that it has been released yet. Yes, once part has been moved to disk, it will be slower than query a purely in-memory version.

At this point, I think it would be helpful to specify exactly what it is you are trying to do.

**Comment by SebCli78** [comment]
my simple goal is to have an efficient and performing way to get utxo list from an address, almost in real time like was possible before using cardano-cli, but not since 1.31.0 if I'm not wrong.

**Comment by erikd** [comment]
Using the example UTxO query and then filtering it for just the address you want is going to be slow in comparison to a custom written query. That means you need to write an optimized query.
